src/sqlfluff/rules/L034.py

```diff
@@ -94,10 +94,24 @@ def _eval(self, context: RuleContext) -> EvalResultType:
             "insert_statement", "set_expression"
         ):
             return None
+        if (
+            len(context.parent_stack) >= 3
+            and context.parent_stack[-3].is_type("insert_statement", "set_expression")
```

Comment:
Should you have a `set_expression` test case?

Reply:
I tried to create one, but couldn't get it to parse.